src/transforms/explode.rs

```diff
@@ -0,0 +1,327 @@
+use crate::{
+    config::{DataType, GenerateConfig, GlobalOptions, TransformConfig, TransformDescription},
+    event::{Event, Value},
+    internal_events::{ExplodeFieldIsNotArray, ExplodeFieldMissing, ExplodeFieldOverwritten},
+    transforms::{FunctionTransform, Transform},
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct ExplodeConfig {
+    #[serde(default)]
+    pub field: Option<String>,
+    #[serde(default)]
+    pub rename_as: Option<String>,
+}
+
+#[derive(Clone, Debug)]
+pub struct Explode {
+    field: String,
+    rename_as: Option<String>,
+}
+
+inventory::submit! {
+    TransformDescription::new::<ExplodeConfig>("explode")
+}
+
+impl GenerateConfig for ExplodeConfig {
+    fn generate_config() -> toml::Value {
+        toml::Value::try_from(Self {
+            field: None,
+            rename_as: None,
+        })
+        .unwrap()
+    }
+}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "explode")]
+impl TransformConfig for ExplodeConfig {
+    async fn build(&self, _globals: &GlobalOptions) -> crate::Result<Transform> {
+        let field = self
+            .field
+            .clone()
+            .unwrap_or_else(|| crate::config::log_schema().message_key().to_string());
+
+        Ok(Transform::function(Explode::new(
+            field,
+            self.rename_as.clone(),
+        )))
+    }
+
+    fn input_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    fn output_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    fn transform_type(&self) -> &'static str {
+        "explode"
+    }
+}
+
+impl Explode {
+    pub fn new(field: String, rename_as: Option<String>) -> Self {
+        Explode { field, rename_as }
+    }
+}
+
+impl FunctionTransform for Explode {
+    fn transform(&mut self, output: &mut Vec<Event>, mut event: Event) {
+        let target_field = if let Some(rename_as) = &self.rename_as {
+            rename_as
+        } else {
+            &self.field
+        };
+        if let Some(value) = event.as_mut_log().remove_prune(&self.field, true) {
+            if let Value::Array(array) = value {
+                if event
+                    .as_mut_log()
+                    .insert(&target_field, Value::Null)
```

Comment:
I think you could just use `.get()` here to avoid the unnecessary insert operation.